pandas/core/generic.py

```diff
@@ -9077,6 +9078,7 @@ def _where(
         name="where",
         name_other="mask",
     )
+    @deprecate_nonkeyword_arguments(version="2.0", allowed_args=["self", "labels"])
```

Comment:
let's allow self, cond, and other here

Reply:
I made some corrections and did 'git push origin shiny-new-feature', but I can't see the pull request is updated. What have I missed?

Reply:
I also changed the above to be like this:
`@deprecate_nonkeyword_arguments(version="2.0", allowed_args=["self", "cond", "other"])`
Is that what you mean? I couldn't find a way to quote the code as you did, so I just copied the code here.

Reply:
yup, that's what I meant :+1:

Reply:
Thanks a lot. I deleted my previous comments as I realized I had a problem with my local git. Now I have added what's new, and the test. Still not sure I understood this issue fully.

Reply:
> Still not sure I understood this issue fully.

Try running:
```python
In [1]: ser = pd.Series([1,2,3])

In [2]: ser.where(ser>1, 0, False)
Out[2]: 
0    0
1    2
2    3
dtype: int64
```

It works fine - however, in a future release, we'd like to force users to pass arguments (apart from `cond` and `other`) by name, i.e. `ser.where(ser>1, 0, inplace=True)`. But before we do that, we need a FutureWarning. The decorator you're adding here should change the above so that this will happen:
```python
In [1]: ser = pd.Series([1,2,3])

In [2]: ser.where(ser>1, 0, False)
<ipython-input-2-7bc00a67a7aa>:1: FutureWarning: In a future version of pandas all arguments of NDFrame.where except for the arguments 'cond' and 'other' will be keyword-only
  ser.where(ser>1, 0, False)
Out[2]: 
0    0
1    2
2    3
dtype: int64
```

Make sure to fetch and merge latest changes for this to work properly though:
```
git fetch upstream
git merge upstream/master
```

Reply:
Thanks a lot for the detailed explanation. I think I understand it now. 
My test should be: 
' s = Series(range(5))
  with tm.assert_produces_warning(FutureWarning):
         s.where(s > 1, 10, False)'
because 'False' was passed in as a non-keyword (my understand of non-keyword is the same as a positional argument, is this correct?), the future warning will be triggered. 
I hope I got it right this time